source/common/network/socket_interface.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include "envoy/config/typed_config.h"
+#include "envoy/network/socket.h"
+#include "envoy/registry/registry.h"
+#include "envoy/server/bootstrap_extension_config.h"
+
+#include "common/singleton/threadsafe_singleton.h"
+
+#include "absl/container/flat_hash_map.h"
+
+namespace Envoy {
+namespace Network {
+
+// Wrapper for SocketInterface instances returned by createBootstrapExtension() which must be
+// implemented by all factories that derive SocketInterfaceBase
+class SocketInterfaceExtension : public Server::BootstrapExtension {
+public:
+  SocketInterfaceExtension(SocketInterface* sock_interface) : sock_interface_(sock_interface) {}
```

Comment:
nit: take/return/store references here as none of these can be null.

Reply:
Done!